dartagnan/src/main/java/com/dat3m/dartagnan/wmm/relation/base/memory/RelRf.java

```diff
@@ -88,86 +90,100 @@ public TupleSet getMaxTupleSet(){
     }
 
     @Override
-    protected BoolExpr encodeApprox(Context ctx) {
-        BoolExpr enc = ctx.mkTrue();
-        Map<MemEvent, List<BoolExpr>> edgeMap = new HashMap<>();
-        Map<MemEvent, BoolExpr> memInitMap = new HashMap<>();
+    protected BooleanFormula encodeApprox(SolverContext ctx) {
+    	FormulaManager fmgr = ctx.getFormulaManager();
+		BooleanFormulaManager bmgr = fmgr.getBooleanFormulaManager();
+    	
+    	BooleanFormula enc = bmgr.makeTrue();
+        Map<MemEvent, List<BooleanFormula>> edgeMap = new HashMap<>();
+        Map<MemEvent, BooleanFormula> memInitMap = new HashMap<>();
 
         boolean canAccNonInitMem = task.getSettings().getFlag(Settings.FLAG_CAN_ACCESS_UNINITIALIZED_MEMORY);
         boolean useSeqEncoding = task.getSettings().getFlag(Settings.FLAG_USE_SEQ_ENCODING_REL_RF);
 
         for(Tuple tuple : maxTupleSet){
             MemEvent w = (MemEvent) tuple.getFirst();
             MemEvent r = (MemEvent) tuple.getSecond();
-            BoolExpr edge = this.getSMTVar(tuple, ctx);
+            BooleanFormula edge = this.getSMTVar(tuple, ctx);
+            IntegerFormula a1 = w.getMemAddressExpr() instanceof BitvectorFormula ? 
```

Comment:
There are so many copies of this code. A "convertToIntegerFormula" method in some Helper class would be better. Basically the new method you used in Memory but refactored into a static Helper class.
Also, the check for Addresses being BitVectors is not done cosistently (a lot of code just assumes Integers)